apps/game_backend/lib/game_backend/campaigns.ex

```diff
@@ -41,10 +37,86 @@ defmodule GameBackend.Campaigns do
     |> Repo.insert()
   end
 
+  @doc """
+  Inserts a campaign.
+  """
+  def insert_campaign(attrs, opts \\ []) do
+    %Campaign{}
+    |> Campaign.changeset(attrs)
+    |> Repo.insert(opts)
+  end
+
+  @doc """
+  Inserts a campaign progress.
+  """
+  def insert_campaign_progress(attrs, opts \\ []) do
+    %CampaignProgress{}
+    |> CampaignProgress.changeset(attrs)
+    |> Repo.insert(opts)
+  end
+
+  @doc """
+  Inserts a super campaign.
+  """
+  def insert_super_campaign(attrs, opts \\ []) do
+    %SuperCampaign{}
+    |> SuperCampaign.changeset(attrs)
+    |> Repo.insert(opts)
+  end
+
   @doc """
   Get a level by id.
+
+  Returns `{:error, :not_found}` if no level is found.
   """
   def get_level(level_id) do
-    Repo.get(Level, level_id) |> Repo.preload(units: :items, units: :character)
+    level = Repo.get(Level, level_id) |> Repo.preload(units: :items, units: :character)
+    if level, do: {:ok, level}, else: {:error, :not_found}
```

Comment:
why are we always wrapping the results of repo getters inside tuples? isn't it easier to just return the value and handle nil or not nil results?

Reply:
This allows are functions to use `with` patterns that make code much more readable! See Unit level_up for example

Reply:
You can still use with statements with `{:ok, entity}` and `nil` returning values.
Also, sometimes the `with` statement is really hard to follow, specially when it comes with an else clause. Use it carefully when it improves the code readability.

Quoting [elixirschool.com](https://elixirschool.com/en/lessons/basics/control_structures):
> The special form with/1 is useful when you might use a nested case/2 statement or situations that cannot cleanly be piped together.